components/mediation-admin/org.wso2.carbon.message.processor/src/main/java/org/wso2/carbon/message/processor/service/MessageProcessorAdminService.java

```diff
@@ -755,4 +761,111 @@ public boolean validateAxis2ClientRepo(String input) throws AxisFault {
+     * @return <code>message</code> returns message received from the queue as a string.
+     */
+    public String getMessage(String processorName) throws AxisFault {
+        SynapseConfiguration configuration = getSynapseConfiguration();
+        MessageConsumer messageConsumer = getMessageConsumer(configuration,processorName);
+        String message = null;
+
+        try {
+            message = getMessageAsString(messageConsumer);
+        } catch (AxisFault e) {
+            handleException(log, "Failed to get message :", e);
+        }
+
+        messageConsumer.cleanup();
+        return message;
+    }
+
+    private String getMessageAsString(MessageConsumer messageConsumer) throws AxisFault {
+        MessageContext messageContext;
+        String message = null;
+
+        if (messageConsumer.isAlive()) {
+            try {
+                messageContext = messageConsumer.receive();
+                message = messageContext.getEnvelope().toString();
+            } catch (SynapseException e) {
+                handleException(log, "Failed to get message :", e);
+            }
+        }
+        return  message;
+    }
+
+    /**
+     * Pops the message from the associated queue.
+     *
+     * @param processorName message processor name.
+     */
+    public void popMessage(String processorName) throws AxisFault {
+        SynapseConfiguration configuration = getSynapseConfiguration();
+        MessageConsumer messageConsumer = getMessageConsumer(configuration,processorName);
+
+        try {
+            popMessageFromQueue(messageConsumer);
+        } catch (SynapseException e) {
+            handleException(log, "Failed to get message : ", e);
+        }
+
+        messageConsumer.cleanup();
+    }
+
+    private  void popMessageFromQueue(MessageConsumer messageConsumer) throws AxisFault {
+        try {
+            messageConsumer.receive();
+            messageConsumer.ack();
+        } catch (SynapseException e) {
+           handleException(log, "Failed to get message :" , e);
+        }
+    }
+
+    /**
+     * Pops the message and redirects it to a specified queue.
+     *
+     * @param processorName message processor name.
+     * @param storeName name of the store to redirect the message.
+     */
+    public void popAndRedirectMessage(String processorName, String storeName) throws AxisFault {
+        SynapseConfiguration configuration = getSynapseConfiguration();
+        MessageConsumer messageConsumer = getMessageConsumer(configuration,processorName);
+        MessageProducer messageProducer = configuration.getMessageStore(storeName).getProducer();
+
+        try {
+            popAndRedirectMessageToStore(messageProducer, messageConsumer);
+        } catch (AxisFault e) {
+            handleException(log, "Failed to redirect message :", e);
+        }
+
+        messageConsumer.cleanup();
+    }
+
+    private void popAndRedirectMessageToStore(MessageProducer messageProducer, MessageConsumer messageConsumer)
```

Comment:
This method is only called by popAndRedirectMessage() right? Would it be better to move the logic into the upper class?

Reply:
I have applied a similar private function call for the other two methods as well since it makes handling the exceptions easier. There isnt any upper class that we can more the functions to.

Reply:
:+1: Btw, I meant to say 'the upper method' :)

Reply:
Moved the messageConsumer.cleanup() to a finally block